Makefile

```diff
@@ -1,25 +1,187 @@
+.DEFAULT_GOAL := help
+########################################################
+#       ONDEWO Standard Make Targets
+########################################################
+
+setup_developer_environment_locally: install_packages install_precommit_hooks
+
+install_packages: ## Install npm packages
+	npm i
+
+install_precommit_hooks: ## Install precommit hooks
+	npx husky install
+
+prettier: ## Checks formatting with Prettier - Use PRETTIER_WRITE=-w to also automatically apply corrections where needed
+	node_modules/.bin/prettier --config .prettierrc --check --ignore-path .prettierignore $(PRETTIER_WRITE) ./
+
+eslint: ## Checks Code Logic and Typing
+	./node_modules/.bin/eslint .
+
+TEST:	## Prints some important variables
+	@echo "Release Notes: \n \n $(CURRENT_RELEASE_NOTES)"
+	@echo "GH Token: \t $(GITHUB_GH_TOKEN)"
+	@echo "NPM Name: \t $(NPM_USERNAME)"
+	@echo "NPM Password: \t $(NPM_PASSWORD)"
+
+help: ## Print usage info about help targets
+	# (first comment after target starting with double hashes ##)
+	@grep -E '^[a-zA-Z_-]+:.*?## .*$$' Makefile | sort | awk 'BEGIN {FS = ":.*?## "}; {printf "\033[36m%-40s\033[0m %s\n", $$1, $$2}'
+
+makefile_chapters: ## Shows all sections of Makefile
+	@echo `cat Makefile| grep "########################################################" -A 1 | grep -v "########################################################"`
+
+########################################################
+#       Repo Specific Make Targets
+########################################################
+
+########################################################
+#		Release
+
+release:
+	@echo "Start Release"
+	make build_and_publish_npm_via_docker
+	make create_release_branch
+	make create_release_tag
+	make release_to_github_via_docker_image
+
+gh_release: build_utils_docker_image release_to_github_via_docker_image
+
+npm_release:
+	@echo "Start NPM Release"
+	npm publish ./npm --access public --dry-run
+	@echo "Finished NPM Release"
+
+create_release_branch: ## Create Release Branch and push it to origin
+	git checkout -b "release/${ONDEWO_NLU_VERSION}"
+	git push -u origin "release/${ONDEWO_NLU_VERSION}"
 
-build: check_out_correct_submodule_versions copy_proto_files_all_submodules npm_run_build
+create_release_tag: ## Create Release Tag and push it to origin
+	git tag -a ${ONDEWO_NLU_VERSION} -m "release/${ONDEWO_NLU_VERSION}"
+	git push origin ${ONDEWO_NLU_VERSION}
 
-check_out_correct_submodule_versions: 
+login_to_gh: ## Login to Github CLI with Access Token
+	echo $(GITHUB_GH_TOKEN) | gh auth login -p ssh --with-token
+
+build_gh_release: ## Generate Github Release with CLI
+	gh release create --repo $(GH_REPO) "$(ONDEWO_NLU_VERSION)" -n "$(CURRENT_RELEASE_NOTES)" -t "Release ${ONDEWO_NLU_VERSION}"
+
+########################################################
+#		Docker
+
+push_to_gh: login_to_gh build_gh_release
+	@echo 'Released to Github'
+
+build_compiler:
+	cd ondewo-proto-compiler/angular && sh build.sh
```

Comment:
Can you please 
1.) upgrade the libraries in the proto compiler (grpc 1.47.0 and the newest protoc)
2.) add "export NG_CLI_ANALYTICS=ci" to suppress questions => https://stackoverflow.com/questions/56355499/stop-angular-cli-asking-for-collecting-analytics-when-i-use-ng-build